Update The-Basics.md

NeilMasters opened the pull request:
Adding a suggested change to ternary operators where its not actually necessary to use them but is commonly seen.

NeilMasters committed:
Update The-Basics.md

Adding a suggested change to ternary operators where its not actually necessary to use them but is commonly seen.

dragoonis commented:
:+1: on this, I too often see people doing:

``` php
return ($a == $b) ? true false
```

instead of 

``` php
return $a == $b;
```

philsturgeon commented on pages/The-Basics.md:
```diff
@@ -304,6 +304,19 @@ vs.
 $a = 5;
 return ($a == 5) ? 'yay' : 'nope';    // this example will return 'yay'
+
+It should be noted to save confusion that you do not need to use a ternary operator for returning a boolean value. An example of this would be.
```
There is a lot of plan-text inside the syntax highlighting block. Can you split the block up?

NeilMasters committed:
Update The-Basics.md

Updated to conform to rest of document and I believe that I have utilized the correct syntax for then highlighters.

NeilMasters commented on pages/The-Basics.md:
```diff
@@ -304,6 +304,25 @@ vs.
+vs
+
+$a = 3;
+return $a == 3; // Will return true or false if $a == 3
```
Should we provide some examples of where to use bracketing for readability? I.e. $a == 3 is fine but what about $a == 3 && $b == 4 && $c == 5 or possibly more confusion a mix of all possibilities $a == 3 && $b === 4 || $c !== 5

philsturgeon replied:
.
-- 
Phil Sturgeon

On March 12, 2014 at 12:27:23 PM, Neil Masters (notifications@github.com) wrote:

In pages/The-Basics.md:

> @@ -304,6 +304,25 @@ vs.
> 
>  $a = 5;
>  return ($a == 5) ? 'yay' : 'nope';    // this example will return 'yay'
> +
> +{% endhighlight %}
> +
> +It should be noted that you do not need to use a ternary operator for returning a boolean value. An example of this would be.
> +
> +{% highlight php %}
> +<?php
> +$a = 3;
> +return ($a == 3) ? true : false; // Will return true or false if $a == 3
> +
> +vs
> +
> +$a = 3;
> +return $a == 3; // Will return true or false if $a == 3
> Should we provide some examples of where to use bracketing for readability? I.e. $a == 3 is fine but what about $a == 3 && $b == 4 && $c == 5 or possibly more confusion a mix of all possibilities $a == 3 && $b === 4 || $c !== 5

—
Reply to this email directly or view it on GitHub.

philsturgeon replied:
Yup :)
-- 
Phil Sturgeon

On March 12, 2014 at 12:27:23 PM, Neil Masters (notifications@github.com) wrote:

In pages/The-Basics.md:

> @@ -304,6 +304,25 @@ vs.
> 
>  $a = 5;
>  return ($a == 5) ? 'yay' : 'nope';    // this example will return 'yay'
> +
> +{% endhighlight %}
> +
> +It should be noted that you do not need to use a ternary operator for returning a boolean value. An example of this would be.
> +
> +{% highlight php %}
> +<?php
> +$a = 3;
> +return ($a == 3) ? true : false; // Will return true or false if $a == 3
> +
> +vs
> +
> +$a = 3;
> +return $a == 3; // Will return true or false if $a == 3
> Should we provide some examples of where to use bracketing for readability? I.e. $a == 3 is fine but what about $a == 3 && $b == 4 && $c == 5 or possibly more confusion a mix of all possibilities $a == 3 && $b === 4 || $c !== 5

—
Reply to this email directly or view it on GitHub.

NeilMasters committed:
Update The-Basics.md

Added section for using brackets for both form and function. I would appreciate a second set of eyes over this to clarify its easy to understand.

philsturgeon commented on pages/The-Basics.md:
```diff
@@ -304,6 +304,77 @@ vs.
+$b = 4;
+$c = 5;
+
+return $a == 3 || $b === 4 || $c !== 5; // return true if $a == 3 or $b === 4(value and type) or $c !== 5(value and type)
```
This is a weird mixture of code and text.

Try and have it as codeblock with any text outside of it. Your `vs` could be a comment, so `// vs`.

philsturgeon commented:
Could you make the formatting changes I suggested?

NeilMasters committed:
Update The-Basics.md

Tidying up the wordyness, removing overused and thus redundant variable assignments and improving explanations in replacement for quick & short comments.

NeilMasters commented:
Updated. I was somewhat confused by your request but made some amendments that irked me a little. If these recent changes are not what you were referencing then let me know.

philsturgeon commented:
Thank you! I finally got this sorted out. Minor tweakage, but it's in. :)
